lib/timers.js

```diff
@@ -763,6 +792,12 @@ function createImmediate(args, callback) {
 exports.clearImmediate = function(immediate) {
   if (!immediate) return;
 
+  if (isPromise(immediate))
+    immediate = immediate.timer;
+
+  if (typeof immediate._onImmediate !== 'function') {
```

Comment:
If the original `immediate` was a promise _not_ coming from `setImmediate()`, `immediate` should be `undefined` here, and this might crash. Not sure if we want that.